packages/circuit-ui/components/ListItemGroup/ListItemGroup.tsx

```diff
@@ -276,13 +293,24 @@ export const ListItemGroup = forwardRef(
           {items.map(({ key, ...item }) => (
             <StyledLi
               key={key}
+              isFocused={focusedItemKey === key}
               isInteractive={isInteractive}
               selected={item.selected}
             >
               <StyledListItem
                 {...item}
                 isPlain={isPlain}
                 aria-pressed={item.onClick ? item.selected : undefined}
+                onFocus={(event) => {
+                  try {
+                    if (event.currentTarget.matches(':focus-visible')) {
+                      setFocusedItemKey(key);
+                    }
+                  } catch (err) {
+                    setFocusedItemKey(key);
+                  }
```

Comment:
The case here is that some browsers don't support the `:focus-visible` pseudo selector and throw an exception. If they do we call `setFocusedItemKey` only if the condition is true. For browsers that don't support `:focus-visible` we call `setFocusedItemKey` in the catch, since we are in the `onFocus` handler. In other words we don't call `setFocusedItemKey` only if the item is focused and not `:focus-visible`.

Reply:
Please let me know if you figure out a way to do this without JS 🙏

Reply:
Thanks! Sounds good. Since I don't have any time to look into this at the moment, let's roll with it and we can see if we find better solutions in the future (which wouldn't necessarily require breaking changes since this is implementation details land)